client/my-sites/checkout/checkout/composite-checkout-container.js

```diff
@@ -19,52 +19,19 @@ import {
  */
 import wp from 'lib/wp';
 
-const initialCart = {
-	coupon: '',
-	currency: 'BRL',
-	is_coupon_applied: false,
-	products: [
-		{
-			extra: {
-				context: 'signup',
-				domain_registration_agreement_url:
-					'https://wordpress.com/automattic-domain-name-registration-agreement/',
-				privacy: true,
-				privacy_available: true,
-				registrar: 'KS_RAM',
-			},
-			free_trial: false,
-			meta: 'asdkfjalsdkjfalsdjkflaksdjflkajsdfffd.com',
-			product_id: 106,
-			volume: 1,
-		},
-		{
-			extra: {
-				context: 'signup',
-				domain_to_bundle: 'asdkfjalsdkjfalsdjkflaksdjflkajsdfffd.com',
-			},
-			free_trial: false,
-			meta: '',
-			product_id: 1009,
-			volume: 1,
-		},
-	],
-	tax: {
-		display_taxes: false,
-		location: {},
-	},
-	temporary: false,
-};
-
 const registry = createRegistry();
 const { registerStore } = registry;
 
 const wpcom = wp.undocumented();
 
-const useShoppingCart = makeShoppingCartHook(
-	( cartKey, cartParam ) => wpcom.setCart( cartKey, cartParam ),
-	initialCart
-);
+export const getServerCart = wpcom.getCart;
+
+const useShoppingCart = cartKey => {
```

Comment:
I think it might be best not to start this function with `use` since it's not a hook, it's a hook creator creator. Maybe something like `buildMakeShoppingCartHook`? 🤔 Although that's pretty repetative. Is it possible to pass `cartKey` as an argument to `makeShoppingCartHook` directly?

This also makes me wonder if it's safe to be re-creating a hook on every render. With the old version of `makeShoppingCartHook`, it was called when the file loaded so there was only ever one version of the actual hook. This version will call the creator creator on every render, which will create a new version of the hook. If the hook has any internal state, I think this might cause problems.

So a better question might be: since `cartKey` is dynamic and changes on render (and is therefore a dependency for any effects of the hook), is it possible to pass `cartKey` as an argument to the hook itself, rather than preparing it here?